frontend/src/server/Room.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+var app = require("express")();
+var http = require("http").Server(app);
+var io = require("socket.io")(http);
+
+const port = 3000;
+
+app.get("/", function(req, res) {
+  res.sendFile(__dirname + "/index.html");
+});
+
+io.on("connection", function(socket) {
+  console.log("a user connected");
+
+  socket.on("join_room", room => {
+    socket.join(room);
+  });
+
+    http.listen(port, function() {
+  console.log(`listening on *:${port}`);
+});
+
+socket.emit("new_visitor", visitor);
```

Comment:
It's a bit hard to follow with the indentation, but is this just emitting without there having necessarily been a new visitor?